dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CensorHelper.java

```diff
@@ -0,0 +1,210 @@
+package io.dockstore.webservice.helpers;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Inspects input Strings for long runs of base64 characters and X's them out if they appear to
+ * be keys.  Useful for logging purposes.
+ *
+ * CensorHelper identifies potential secrets as longer substrings composed of non-padding base64
+ * characters.  Such substrings could be keys (AWS keys and GitHub tokens are 40-character base64
+ * encoded strings) or other sensitive information, but they could also be other information made
+ * up of the same characters, like file paths, that should be logged.  For example, the path
+ * "/MyStuff/data/input123" is composed entirely of non-padding base64 characters, but is not a
+ * secret and should not be censored.
+ *
+ * So, when CensorHelper finds a longer base64-ish substring, it censors it if all the characters
+ * are hex/decimal digits, and otherwise performs a statistical analysis to determine if the
+ * substring is an encoded key versus something else.  The current implementation combines two
+ * measures of the substring:
+ *
+ * 1. The number of adjacent characters of the same class, where "class" is either a) decimal
+ *    digits, b) uppercase letters, c) lowercase letters + punctuation.
+ *
+ * 2. The number of "English points", where higher values represent sequences of characters more
+ *    characteristic of English language sequences, currently derived from letter-triplet
+ *    frequencies as calculated from an English text corpus.  Passages that resemble English text
```

Comment:
> Trying to wrap my mind around this, how is this whole thing going to react in general to non-English secrets or secrets with accented characters?

The current implementation will censor longer base64-ish strings that look like keys/tokens (such as AWS keys and GitHub tokens), it's not designed to censor anything else.

Reply:
But if I understand correctly, if this is using similarity to English to determine what should not be censored, wouldn't non-english text just look like random text and be more likely to be censored? 
i.e. log messages with non-english names, users with non-english names, etc.

Reply:
The basic algorithm is something like:

1. Find "suspect" strings composed entirely of non-padding base64 (letter, numbers, `-`, `_`, `+`, and `/`) characters, of length 22 or more.
2. For each "suspect" string, calculate its score, and it's below a certain threshold, censor.

So, yeah, if there's a string that matches the criteria in Step 1 and doesn't look like English _at all_, it's likely to be censored.        Letter-frequency-wise, English is at least somewhat similar to many European languages, so candidate strings that are Spanish/French/German/etc will probably not be censored much of the time.  Strings that contain the higher portion of Latin-8 (common western characters with accents, etc) or the non-ASCII portion of Unicode (Asian characters, etc) will never be censored, because those aren't non-padding base64 characters, the strings won't be "suspects".

Reply:
Oh, I see. So something needs to pass `BASE64_PATTERN` to be considered for censorship. 

May want to add
```
        confirmCensored("export BUNDLE_ENTERPRISE__CONTRIBSYS__COM=cafebabe:deadbeef");
        confirmCensored("export BUNDLE_ENTERPRISE__CONTRIBSYS__COM=secretPassword");
```
https://github.com/gitleaks/gitleaks which also has something a bit more off-the-shelf https://github.com/gitleaks/gitleaks/blob/master/config/gitleaks.toml

Reply:
> Oh, I see. So something needs to pass `BASE64_PATTERN` to be considered for censorship.
> 
> May want to add
> 
> ```
>         confirmCensored("export BUNDLE_ENTERPRISE__CONTRIBSYS__COM=cafebabe:deadbeef");
>         confirmCensored("export BUNDLE_ENTERPRISE__CONTRIBSYS__COM=secretPassword");
> ```
> 
> https://github.com/gitleaks/gitleaks which also has something a bit more off-the-shelf https://github.com/gitleaks/gitleaks/blob/master/config/gitleaks.toml

`CensorHelper` won't censor either one of these strings, they're too regular and resemble English.